src/esp/gfx/CubeMap.cpp

```diff
@@ -34,14 +34,17 @@ const Mn::GL::Framebuffer::ColorAttachment rgbaAttachment =
     Mn::GL::Framebuffer::ColorAttachment{0};
 const Mn::GL::Framebuffer::ColorAttachment objectIdAttachment =
     Mn::GL::Framebuffer::ColorAttachment{1};
+// vsm = variance shadow map
+const Mn::GL::Framebuffer::ColorAttachment vsmAttachment =
+    Mn::GL::Framebuffer::ColorAttachment{2};
 
 /**
  * @brief check if the class instance is created with corresponding texture
  * enabled
  */
-void textureTypeSanityCheck(CubeMap::Flags& flag,
-                            CubeMap::TextureType type,
-                            const std::string& functionNameStr) {
+void textureTypeSanityCheck(const std::string& functionNameStr,
```

Comment:
I might be overdoing the optimizations, but since the string is used only for assertions and thus should be as zero-overhead as possible, maybe go with just

```suggestion
void textureTypeSanityCheck(const char* functionNameStr,
```

instead?

Reply:
Ha, thanks! Good suggestions.